### PR TITLE
feat(athena): Shift+Click opens in right sidebar, not just Shift+Enter

### PR DESCRIPTION
**Pull request by agentydragon**
Currently Athena only respects Shift modifier as "open in sidebar" with Enter key, not with left click.

Tested:

* Shift-Enter
* Shift-LeftClick

Not implementing this for new pages yet, because dispatching the open effect for them needs to be delayed until db is updated, and that'd make the code a bit messier.

### CODE REVIEW

File: src/cljs/athens/views/athena.cljs [hunk 1]
```diff
@@ -316,11 +325,11 @@
...
                                                   ^{:key i}
-                                                  [:div (use-style result-style {:on-click (fn [_]
-                                                                                             (let [uid (gen-block-uid)]
-                                                                                               (dispatch [:athena/toggle])
-                                                                                               (dispatch [:page/create query uid])
-                                                                                               (navigate-uid uid)))
+                                                  [:div (use-style result-style {:on-click #(let [uid (gen-block-uid)]
+                                                                                              (dispatch [:athena/toggle])
+                                                                                              (dispatch [:page/create query uid])
+                                                                                               ;; TODO(agentydragon): Open the new page in sidebar if Shift is pressed
```
Comment (neotyk) [inline src/cljs/athens/views/athena.cljs]: Is this TODO left on purpose?

File: src/cljs/athens/views/athena.cljs [hunk 2]
```diff
@@ -316,11 +325,11 @@
...
-                                                                                             (let [uid (gen-block-uid)]
-                                                                                               (dispatch [:athena/toggle])
-                                                                                               (dispatch [:page/create query uid])
-                                                                                               (navigate-uid uid)))
+                                                  [:div (use-style result-style {:on-click #(let [uid (gen-block-uid)]
+                                                                                              (dispatch [:athena/toggle])
+                                                                                              (dispatch [:page/create query uid])
+                                                                                              ;; TODO(agentydragon): Open the new page in sidebar if Shift is pressed
+                                                                                              (navigate-uid uid))
```
Comment (tangjeff0) [inline src/cljs/athens/views/athena.cljs:332]: Not necessary to switch from fn to anonymous function

File: src/cljs/athens/views/athena.cljs [hunk 3]
```diff
@@ -248,24 +248,33 @@
     [:span "Find or Create a Page"]]])
 
 
+(defn open-in-main-or-sidebar
+  "Opens the given uid either in the main area, or in the sidebar, depending
+  on whether Shift is pressed."
+  [e uid]
+  (if (.. e -shiftKey)
+    (dispatch [:right-sidebar/open-item uid])
+    (navigate-uid uid)))
```
Comment (tangjeff0) [inline src/cljs/athens/views/athena.cljs:257]: Unnecessary function, see later comment

File: src/cljs/athens/views/athena.cljs [hunk 4]
```diff
@@ -330,14 +339,14 @@
...
+                                                                                 :on-click (fn [e]
                                                                                              (let [selected-page {:node/title   title
                                                                                                                   :block/uid    uid
                                                                                                                   :block/string string
                                                                                                                   :query        query}]
                                                                                                (dispatch [:athena/toggle])
                                                                                                (dispatch [:athena/update-recent-items selected-page])
-                                                                                               (navigate-uid uid)))
+                                                                                               (open-in-main-or-sidebar e uid)))
```
Comment (tangjeff0) [inline src/cljs/athens/views/athena.cljs:349]: You can just use `navigate-uid` here, and pass the `e` parameter to it.

**Review by tangjeff0 — CHANGES_REQUESTED** [review]

File: src/cljs/athens/views/athena.cljs (hunk 1 above)
Comment (agentydragon) [reply src/cljs/athens/views/athena.cljs]: yes, this PR does not work for new pages. I might get around to fixing that but the PR is probably already useful like this.

File: src/cljs/athens/views/athena.cljs (hunk 4 above)
Comment (agentydragon) [reply src/cljs/athens/views/athena.cljs:349]: oh, nice, I didn't check / know about that.

**Comment by agentydragon** [comment]
Thanks @tangjeff0

**Review by tangjeff0 — APPROVED** [review]
Nice job @agentydragon !

**Pull request merged**